stl/inc/tuple

```diff
@@ -1079,23 +1079,48 @@ constexpr decltype(auto) apply(_Callable&& _Obj, _Tuple&& _Tpl) noexcept(
         make_index_sequence<tuple_size_v<remove_reference_t<_Tuple>>>{});
 }
 
+#if _HAS_CXX20
 #if _HAS_CXX23
 template <class _Ty, _Tuple_like _Tuple, size_t... _Indices>
 #else // ^^^ _HAS_CXX23 / !_HAS_CXX23 vvv
 template <class _Ty, class _Tuple, size_t... _Indices>
 #endif // ^^^ !_HAS_CXX23 ^^^
-constexpr _Ty _Make_from_tuple_impl(_Tuple&& _Tpl, index_sequence<_Indices...>) noexcept(
-    is_nothrow_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD forward<_Tuple>(_Tpl)))...>) {
-    // construct _Ty from the elements of _Tpl
-    static_assert(is_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD forward<_Tuple>(_Tpl)))...>,
-        "the target type must be constructible from the fields of the argument tuple (N4950 [tuple.apply]/4).");
+    requires is_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD declval<_Tuple>()))...>

```

Comment:
Can we just reuse `_Can_make_from_tuple<_Ty, _Tuple, index_sequence<_Indices...>>` here, since it should have been already instantiated?

Reply:
Done, reuse  reuse `_Can_make_from_tuple` now, but I'm not sure whether we should do that since `is_constuctible_v` is closer to the wording of the standard